packages/salesforce-adapter/test/filters/add_parent_to_metadata_instances_within_folder.test.ts

```diff
@@ -0,0 +1,85 @@
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+
+import { CORE_ANNOTATIONS, Element, ReferenceExpression } from '@salto-io/adapter-api'
+import { buildElementsSourceFromElements } from '@salto-io/adapter-utils'
+import { mockTypes } from '../mock_elements'
+import { createInstanceElement } from '../../src/transformers/transformer'
+import { FilterWith } from './mocks'
+import filterCreator from '../../src/filters/add_parent_to_metadata_instances_within_folder'
+import { defaultFilterContext } from '../utils'
+import { buildFetchProfile } from '../../src/fetch_profile/fetch_profile'
+
+describe('addParentToMetadataInstancesWithinFolder filter', () => {
+  let elements: Element[]
+  let filter: FilterWith<'onFetch'>
+  let emailTemplateInstance: Element
+  let emailFolderInstance: Element
+  let reportInstance: Element
+  let reportFolderInstance: Element
+  let documentInstance: Element
+  let documentFolderInstance: Element
+  let dashboardInstance: Element
+  let dashboardFolderInstance: Element
+
+  beforeEach(() => {
+    elements = [
+      (emailTemplateInstance = createInstanceElement(
+        { fullName: 'MarketingFolder/WelcomeEmail' },
+        mockTypes.EmailTemplate,
+      )),
+      (emailFolderInstance = createInstanceElement({ fullName: 'MarketingFolder' }, mockTypes.EmailFolder)),
+      (reportInstance = createInstanceElement({ fullName: 'SalesFolder/QuarterlySales' }, mockTypes.Report)),
+      (reportFolderInstance = createInstanceElement({ fullName: 'SalesFolder' }, mockTypes.ReportFolder)),
+      (documentInstance = createInstanceElement({ fullName: 'SharedDocs/Policy' }, mockTypes.Document)),
+      (documentFolderInstance = createInstanceElement({ fullName: 'SharedDocs' }, mockTypes.DocumentFolder)),
+      (dashboardInstance = createInstanceElement({ fullName: 'Dashboards/PerformanceDashboard' }, mockTypes.Dashboard)),
+      (dashboardFolderInstance = createInstanceElement({ fullName: 'Dashboards' }, mockTypes.DashboardFolder)),
+      ...Object.values(mockTypes),
+    ]
+    filter = filterCreator({
+      config: {
+        ...defaultFilterContext,
+        fetchProfile: buildFetchProfile({ fetchParams: { target: [] } }),
+        elementsSource: buildElementsSourceFromElements(elements),
+      },
+    }) as typeof filter
+  })
+
+  describe('addParentToMetadataInstancesWithinFolder onfetch', () => {
+    describe('when the instances are within folder', () => {
+      beforeEach(async () => {
+        await filter.onFetch(elements)
+      })
+      it('should add parent annotation to email template', async () => {
+        expect(emailTemplateInstance.annotations[CORE_ANNOTATIONS.PARENT]).toBeDefined()
+        expect(emailTemplateInstance.annotations[CORE_ANNOTATIONS.PARENT][0]).toEqual(
+          new ReferenceExpression(emailFolderInstance.elemID, emailFolderInstance),
+        )
```

Comment:
```suggestion
        expect(emailTemplateInstance.annotations[CORE_ANNOTATIONS.PARENT).toEqual([
          new ReferenceExpression(emailFolderInstance.elemID, emailFolderInstance),
        ])
```
nit: Consider changing in other UTs as-well